### ISSUE TITLE
jfrog cli doesn't account for go mod replacements

### ISSUE THREAD

**Issue by darwish** [jfrog/jfrog-cli-go#262]
If a dependency in the `go.mod` file has a replacement, the new path and version are ignored by the jfrog cli, and the old one is published instead.

You can see this clearly by adding a dependency without a hostname prefix:

```
module mycompany/projectA

require (
	mycompany/projectB v0.0.0
)

replace mycompany/projectB => gitlab.mycompany.com/mycompany/projectB v0.0.0-20181101000000-abcdefghi
```

This results in the following output when running `jfrog rt go build go`:

```
[Debug] The following file is missing: /home/go/pkg/mod/cache/download/mycompany/projectB/@v/v0.0.0.zip
```

Because of this, the cli will always think that there are dependencies missing in go-local, and will therefore always attempt to download and publish them.

**Comment by AlexeiVainshtein** [comment]
Thanks for the PR @darwish. This is much appreciated. I just wanted to let you know that this issue was already fixed with this PR #266 and [this](https://github.com/jfrog/jfrog-cli-go/blob/master/jfrog-cli/artifactory/utils/golang/project/dependencies/dependenciesutils.go#L436) is the function in the code. CLI gets the dependencies values from the mod file directly.
This was already released with the latest CLI (version 1.22.0). 
We will appreciate your feedback.

**Comment by darwish** [comment]
Thanks for the update. Glad the issue is resolved.
